arangod/RestHandler/RestAdminLogHandler.cpp

```diff
@@ -157,14 +160,20 @@ void RestAdminLogHandler::reportLogs(bool newFormat) {
 
       network::RequestOptions options;
       options.timeout = network::Timeout(30.0);
+      options.database = _request->databaseName();
 
       auto f =
           network::sendRequest(pool, "server:" + serverId, fuerte::RestVerb::Get,
                                _request->requestPath(),
                                VPackBuffer<uint8_t>{}, options, buildHeaders(_request->headers()));
-      f.get();
 
-      generateResult(rest::ResponseCode::OK, f.result()->slice());
+      network::Response const& r = f.get();
+      if (r.fail()) {
+        generateError(network::fuerteToArangoErrorCode(r));
+      } else {
+        generateResult(rest::ResponseCode::OK, f.result()->slice());
```

Comment:
Can also be simplified slighty: we already resolved the promise, so instead of `f.result()` we can use `r` directly.
```suggestion
        generateResult(rest::ResponseCode::OK, r.slice());
```